src/chain.js

```diff
@@ -11,19 +11,25 @@ var map = require('./map');
  *
  * Dispatches to the `chain` method of the second argument, if present,
  * according to the [FantasyLand Chain spec](https://github.com/fantasyland/fantasy-land#chain).
+ * Treats the second argument as a FantasyLand Chain, when the the second argument is a function
+ * of the form `a -> x -> b`.
```

Comment:
Did you notice that when I flipped the arguments I could compile the Haskell program and produce the desired value? I'm happy to document the behaviour, but we should fix the bug in our implementation first. ;)

Here's a valid implementation:

```javascript
//  Function$prototype$chain :: (a -> b) ~> (b -> a -> c) -> (a -> c)
var Function$prototype$chain = function(f) {
  var chain = this;
  return function(x) { return f(chain(x))(x); };
};
```

I'll submit a pull request shortly.